src/main/java/climateControl/customGenLayer/GenLayerSubBiome.java

```diff
@@ -39,6 +42,15 @@ public GenLayerSubBiome(long p_i45479_1_, GenLayer biomes, GenLayer rivers, SubB
         this.subBiomeChooser = subBiomeChooser;
         this.mBiomes = mBiomes;
         this.initChunkSeed(0, 0);
+        try {
+            if (doBoP) {
+                BoPSubBiomeReplacer = new BoPSubBiomeReplacer(randomCallback);
+                logger.info("Bop set up");
+            }
+        } catch (java.lang.NoClassDefFoundError e) {
+            BoPSubBiomeReplacer = null;
+            logger.info("no bop ");
+        }
```

Comment:
Replace this with a cached Loader.isModLoaded check

Reply:
would you be able to point me to an example of that?

Reply:
Firstly, create static field and initialize it at pre-init phase:
```java
    private static boolean isBoPLoaded;
    // then at pre-init:
    isBoPLoaded =  = Loader.isModLoaded("BiomesOPlenty");
```
Secondly, replace this code with:
```suggestion
        if (isBoPLoaded && doBoP) {
            BoPSubBiomeReplacer = new BoPSubBiomeReplacer(randomCallback);
        }
```

Reply:
Hi @not-nocturnal do you want to finish up this PR so it can be accepted please?

Reply:
sorry, this ended up being too much for me to get my head around and i burned out trying to finish it

Reply:
> sorry, this ended up being too much for me to get my head around and i burned out trying to finish it

just the check? or what were you stuck on?

Reply:
just the check, but it's been a very long time and i feel very out of depth

Reply:
@OneEyeMaker would you like to pick this up?